modules/rosetta/src/main/scala/org/tessellation/rosetta/server/model/dag/schema.scala

```diff
@@ -0,0 +1,95 @@
+package org.tessellation.rosetta.server.model.dag
+
+import derevo.cats.{eqv, show}
+import derevo.derive
+import enumeratum.{Enum, EnumEntry}
+import io.circe.Encoder
+import io.estatico.newtype.macros.newtype
+
+/** Represents schema changes specific to this project related to metadata or type Any fields which do not properly code generate
+  */
+object schema {
+
+  case class GenericMetadata()
+
+  @derive(eqv, show)
+  sealed trait ChainObjectStatus extends EnumEntry
+
+  object ChainObjectStatus extends Enum[ChainObjectStatus] with ChainObjectStatusCodecs {
+    case object Unknown extends ChainObjectStatus
+    case object Pending extends ChainObjectStatus
+    case object Accepted extends ChainObjectStatus
+
+    override def values: IndexedSeq[ChainObjectStatus] = findValues
+  }
+
+  trait ChainObjectStatusCodecs {
+    implicit val jsonEncoder: Encoder[ChainObjectStatus] =
+      Encoder.forProduct1("status")(_.toString)
+  }
+
+  @newtype
+  case class NetworkChainObjectStatus(value: ChainObjectStatus)
+
+  case class AccountBalanceResponseMetadata(sequenceNum: Int)
+
+  case class AccountCoinsResponseMetadata(sequenceNum: Int)
+
+  case class AccountIdentifierMetadata(alias: Option[String])
+
+  case class AmountMetadata()
+
+  case class BlockMetadata()
+
+  case class CallRequestParametersExampleEndpoint1(
+    paramOne: String
+  )
+
+  case class CallRequestParametersExampleEndpoint2(
+    paramOne: Int
+  )
+
+  case class CallRequestParameters(
+    callRequestParametersExampleEndpoint1: Option[CallRequestParametersExampleEndpoint1],
+    callRequestParametersExampleEndpoint2: Option[CallRequestParametersExampleEndpoint2]
+  )
+
+  case class CallResponseExampleEndpoint1(
+    valueOne: String
+  )
+
+  case class CallResponseExampleEndpoint2(
+    valueOne: Int
+  )
+
+  case class CallResponseActual(
+  )
+
+  case class ConstructionDeriveRequestMetadata()
+
+  case class ConstructionDeriveResponseMetadata()
+
+  case class ConstructionMetadataRequestOptions()
+
+  case class ConstructionMetadataResponseMetadata(
+    lastTransactionHashReference: String,
+    lastTransactionOrdinalReference: Long
+  )
+
+  case class ConstructionParseResponseMetadata()
+
+  case class ConstructionPayloadsRequestMetadata(
+    srcAddress: String,
+    lastTransactionHashReference: String,
+    lastTransactionOrdinalReference: Long,
+    fee: Long,
+    salt: Option[Long]
+  )
```

Comment:
what's going to happen with that class later?
is it an input from a user? if so, can it be already typed properly? if no, are you going to map it into domain classes?

Reply:
This is not being used. The parameters have been removed, but the case class will be left in as a stub in case we need to implement this metadata.

Reply:
@cngo-github removed these -- this has been replaced by the args on the other Metadata class (hash reference and ordinal.) I think we might need some of this in the future but it's not yet required for the test so we can re-add later if necessary.

This is all data that goes to external integrations -- the purpose behind these fields was to provide state dependencies associated with a transaction -- so a user can query to pre-process data. It is returned directly to the user as string types (and then later passed as another input requiring validation.) 

Yes -- all of this data needs to be mapped to domain classes since a user can override it with their own inputs -- meaning it needs to undergo validation. It only acts as an endpoint to provide state related information required for constructing transaction (but is not *required* to be used by the user since they can supply anything, hence requires validation / parsing.)